core/src/heaviest_subtree_fork_choice.rs

```diff
@@ -525,6 +525,26 @@ impl HeaviestSubtreeForkChoice {
         );
     }
 
+    fn heaviest_slot_on_same_voted_fork(&self, tower: &Tower) -> Option<Slot> {
+        tower.last_voted_slot().map(|last_voted_slot| {
+            let heaviest_slot_on_same_voted_fork = self.best_slot(last_voted_slot);
```

Comment:
Add a test that `is_stray_last_vote()` should be checked only after `self.best_slot() -> None`...

Reply:
done